api/types/provisioning.go

```diff
@@ -547,6 +553,20 @@ func ProvisionTokensFromV1(in []ProvisionTokenV1) []ProvisionToken {
 	return out
 }
 
+// ProvisionTokensFromStatic converts static tokens to resource list
+func ProvisionTokensFromStatic(in []ProvisionTokenV1) []ProvisionToken {
```

Comment:
Ok, I added the `IsStatic` method but I am not relying on the time anymore (it seemed fickle to me from the start). When we fetch static tokens, we create a new `ProvisionToken` from the available static tokens. I've just set their origin to `OriginConfiguration` instead of being nil and now `IsStatic` just checks if `Origin() == OriginConfiguration`. It seems to be more correct and resilient this way. Let me know what you think

I've added this method to replace `ProvisionTokensFromV1` and all instances that I could find using the old method had to deal with static tokens. I could have just thrown in this `SetOrigin` there but I was worried it would be used for something that _isnt_ a static token. If that isn't the case, I'll remove this new `ProvisionTokensFromStatic` method and throw it into the old one. Curious your thoughts

Reply:
I think that's fine, but it would be better if we changed it at its creation instead of when converting to something else.
So, either on their CheckAndSetDefaults or when parsing the token from the file.
https://github.com/gravitational/teleport/blob/67a3d64f381393bb6eba42c400a8415713c10117/api/types/statictokens.go#L41
https://github.com/gravitational/teleport/blob/67a3d64f381393bb6eba42c400a8415713c10117/lib/config/fileconf.go#L950